src/azul/chalice.py

```diff
@@ -93,6 +95,12 @@ def authentication_middleware(self, event, get_response):
             response = get_response(event)
         return response
 
+    def _lambda_context_middleware(self, event, get_response):
+        if self.lambda_context is not None and not self.unit_test:
+            remaining_millis = self.lambda_context.get_remaining_time_in_millis()
+            config.is_api_gateway_context = remaining_millis < 30000
```

Comment:
Note https://github.com/DataBiosphere/azul/blob/198b89b68342e14c8c4235e44828893dbcc8cd2f/lambdas/service/.chalice/config.json.template.py#L24

To reliably detect API Gateway we could have two middlewares, one registered for `http` and the other for `all` (the `http` one should run first). But I think I like a heuristic better. We should store the entire context in `config.lambda_context` and move the heuristic to `config.terra_client_timeout`. If the remaining execution time is less than or equal to config.api_gateway_timeout + config.api_gateway_timeout_padding we use the short timeout. We should also word the documentation accordingly, so that it's clear that the short timeout is used whenever the timing is severely constrained, and that API Gateway is one example where it is.